pkg/kube/container_files.go

```diff
@@ -56,6 +56,23 @@ func (c *Container) bundlePath() string {
 	return filepath.Join(contInfoPath, c.id, contBundlePath)
 }
 
+func (c *Container) addLogDirectory() error {
+	logDir := c.pod.GetLogDirectory()
+	logPath := c.GetLogPath()
+	if logDir == "" || logPath == "" {
+		return nil
+	}
+
+	logPath = filepath.Join(logDir, logPath)
+	logDir = filepath.Dir(logPath)
+	err := os.MkdirAll(logDir, os.ModePerm)
```

Comment:
By precaution it would be good to wrap mkdir/mkdirall/openfile by resetting umask and restore it with defer :

```
oldmask := syscall.Umask(0)
defer syscall.Umask(oldmask)
```

To avoid surprise with permissions

Reply:
After slack discussion, just set umask(0) before starting CRI server is simpler

Reply:
Set umask in main.go